mitosheet/mitosheet/user/location.py

```diff
@@ -118,4 +118,13 @@ def is_jupyterlite() -> bool:
     except ImportError:
         pass
 
-    return False
\ No newline at end of file
+    return False
+
+def is_streamlit() -> bool:
+    try:
+        # TODO: have to handle versions of streamlit that don't have this
+        # or it's in different places: https://discuss.streamlit.io/t/how-to-check-if-code-is-run-inside-streamlit-and-not-e-g-ipython/23439/8
+        from streamlit.runtime.scriptrunner import get_script_run_ctx
```

Comment:
We should create out a TODO list testing release procedure for different cases like this. And we should also make sure that we have good logging so we can identify that there are errors and can fix them

Reply:
If we have tests for streamlit (which we will), then we will catch this. I wonder what version of streamlit we should go back to.